bindings/typescript/src/ballot.ts

```diff
@@ -0,0 +1,135 @@
+  }
+
+  /**
+   * @brief The unique ballot id that is meaningful to the consuming application.
+   */
+  get objectId(): string {
+    return this._handle.getObjectId();
+  }
+
+  /**
+   * @brief The Object Id of the ballot style in the election manifest.  This value is used to determine which contests to expect on the ballot, to fill in missing values, and to validate that the ballot is well-formed
+   */
+  get styleId(): string {
+    return this._handle.getObjectId();
+  }
+
+  /**
+   * Export the ballot representation as JSON
+   */
+  toJson(): string {
+    return this._handle.toJson();
+  }
+
+  /**
+   * @brief Creates a PlaintextBallot object from a [RFC-8259](https://www.rfc-editor.org/rfc/rfc8259.html#section-8.1) UTF-8 encoded JSON string
+   * @param json the JSON string
+   */
+  static async fromJson(json: string): Promise<PlaintextBallot> {
+    var result = (await getInstance()).PlaintextBallot.fromJson(json);
+    return new PlaintextBallot(result);
+  }
+}
+
+/**
+ * A CiphertextBallot represents a voters encrypted selections for a given ballot and ballot style.
+ *
+ * When a ballot is in it's complete, encrypted state, the `nonce` is the seed nonce
+ * from which all other nonces can be derived to encrypt the ballot.  Allong with the `nonce`
+ * fields on `Ballotcontest` and `BallotSelection`, this value is sensitive.
+ *
+ * Don't make this directly. Use `make_ciphertext_ballot` instead, or call `eg_encrypt_ballot` in encrypt.h.
+ */
+export class CiphertextBallot {
+  _handle: CiphertextBallotHandle;
+
+  constructor(handle: CiphertextBallotHandle) {
+    this._handle = handle;
+  }
+
+  /**
+   * @brief The unique ballot id that is meaningful to the consuming application.
+   */
+  get objectId(): string {
+    return this._handle.getObjectId();
+  }
+
+  /**
+   * @brief The Object Id of the ballot style in the election manifest.  This value is used to determine which contests to expect on the ballot, to fill in missing values, and to validate that the ballot is well-formed
+   */
+  get styleId(): string {
+    return this._handle.getObjectId();
+  }
+
+  /**
+   * @brief Hash of the complete Election Manifest to which this ballot belongs
+   */
+  get manifestHash(): ElementModQ {
+    var result = this._handle.getManifestHash();
+    return new ElementModQ(result);
+  }
+
+  /**
+   * @brief The unique ballot code for this ballot that is derived from the ballot seed, the timestamp, and the hash of the encrypted values
+   */
+  get ballotCode(): ElementModQ {
+    var result = this._handle.getBallotCode();
+    return new ElementModQ(result);
+  }
+
+  // TODO: get state
+
+  /**
+   * A helper function to mark the ballot as cast and remove sensitive values like the nonce.
+   */
+  cast(): void {
+    this._handle.cast();
+  }
+
+  /**
+   * A helper function to mark the ballot as spoiled and remove sensitive values like the nonce.
+   */
+  spoil(): void {
+    this._handle.spoil();
+  }
```

Comment:
add the new challenged state here.

Reply:
will add it as part of the next wasm PR.